src/web/app/pages-instructor/instructor-session-individual-extension-page/individual-extension-date-modal/individual-extension-date-modal.component.ts

```diff
@@ -0,0 +1,143 @@
+import { SimpleModalService } from '../../../../services/simple-modal.service';
+import { TimezoneService } from '../../../../services/timezone.service';
+import { DateFormat } from '../../../components/datepicker/datepicker.component';
+import { SimpleModalType } from '../../../components/simple-modal/simple-modal-type';
+import { TimeFormat } from '../../../components/timepicker/timepicker.component';
+
+export enum RadioOptions {
+  EXTEND_TO = 1,
+  EXTEND_BY = 2,
+}
+
+/**
+ * Modal to confirm permanent deletion of a feedback session.
+ */
+@Component({
+  selector: 'tm-individual-extension-date-modal',
+  templateUrl: './individual-extension-date-modal.component.html',
+  styleUrls: ['./individual-extension-date-modal.component.scss'],
+})
+
+export class IndividualExtensionDateModalComponent {
+  @Input()
+  numberOfStudents: number = 0;
+
+  @Input()
+  numberOfInstructors: number = 0;
+
+  @Input()
+  feedbackSessionEndingTime: number = 0;
+
+  @Input()
+  feedbackSessionTimeZone: string = '';
+
+  @Output() onConfirmCallBack: EventEmitter<number> = new EventEmitter();
+
+  constructor(public activeModal: NgbActiveModal,
+              private timeZoneService: TimezoneService,
+              private simpleModalService: SimpleModalService) {}
+
+  RadioOptions: typeof RadioOptions = RadioOptions;
+  radioOption: RadioOptions = RadioOptions.EXTEND_BY;
+  extendByDeadlineKey: String = '';
+  extendByDeadlineOptions: Map<String, Number> = new Map([
+    ['12 hours', 0.5],
+    ['1 day', 24],
+    ['3 days', 72],
+    ['1 week', 168],
+    ['Customize', 0],
+  ]);
+
+  DATETIME_FORMAT: string = 'd MMM YYYY h:mm:ss';
+  datePicker: DateFormat = { year: 0, month: 0, day: 0 };
+  timePicker: TimeFormat = { hour: 23, minute: 59 };
+  extendByDatePicker = { minutes: 0, hours: 0, days: 0 };
+
+  onConfirm(): void {
+    if (this.getExtensionTimestamp() < Date.now()) {
+      const extensionTimeString = this.timeZoneService
+      .formatToString(this.getExtensionTimestamp(), this.feedbackSessionTimeZone, this.DATETIME_FORMAT);
+      const currentTimeString = this.timeZoneService
+      .formatToString(Date.now(), this.feedbackSessionTimeZone, this.DATETIME_FORMAT);
+      this.simpleModalService.openConfirmationModal(
+        'Are you sure you want to extend to before the current time?',
+        SimpleModalType.DANGER,
+        `The current time now is ${currentTimeString}`
+        + ` and you are extending to ${extensionTimeString} in ${this.feedbackSessionTimeZone}.`
+        + ' Do you wish to proceed?',
+      ).result.then(() => this.onConfirmCallBack.emit(this.getExtensionTimestamp()));
+    } else {
+      this.onConfirmCallBack.emit(this.getExtensionTimestamp());
+    }
+  }
+
+  onChangeDateTime(data: DateFormat | TimeFormat, field: string): void {
+    if (field === 'date') {
+      this.datePicker = data as DateFormat;
+    }
+    if (field === 'time') {
+      this.timePicker = data as TimeFormat;
+    }
```

Comment:
Should we use something like `if-else if-else` instead of these fallthrough looking blocks. `if-else if` shows the dichotomy better I feel:)